lib/rules/no-wait-for-side-effects.ts

```diff
@@ -86,19 +107,35 @@ export default createTestingLibraryRule<Options, MessageIds>({
       }
     }
 
-    function reportImplicitReturnSideEffect(node: TSESTree.CallExpression) {
+    function reportImplicitReturnSideEffect(
+      node:
+        | TSESTree.CallExpression
+        | TSESTree.AssignmentExpression
+        | TSESTree.SequenceExpression
+    ) {
       if (!isCallerWaitFor(node)) {
         return;
       }
 
-      const expressionIdentifier = getPropertyIdentifierNode(node.callee);
-      if (!expressionIdentifier) {
-        return;
-      }
+      const expressionIdentifier =
+        isCallExpression(node) && getPropertyIdentifierNode(node.callee);
+      const isRenderInAssignment =
+        isAssignmentExpression(node) &&
+        helpers.isRenderUtil(getPropertyIdentifierNode(node.right));
```

Comment:
```suggestion
      const isRenderInAssignment =
        isAssignmentExpression(node) &&
        helpers.isRenderUtil(getPropertyIdentifierNode(node.right));
      if (!isRenderInAssignment) {
        return;
      }
```

Reply:
Early returns worked for `getSideEffectNodes` function, doesn't seem like we can introduce them here 🤔